### PR TITLE
Set with the same digits of precision for mysql timestamp default value

### LINKED ISSUE #2991: liquibase.exception.DatabaseException: Invalid default value for 'created_on' [Failed SQL: ALTER TABLE meeting.sys_event_store MODIFY created_on timestamp(3) DEFAULT current_timestamp()
**Describe the bug**
I try it use database mysql 5.6, mysql 8.0.

all 1.2.0 database update liquibase file like below:

<databaseChangeLog
        xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
                      http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.0.xsd"
        logicalFilePath="KapuaDB/changelog-event-store-1.2.0.xml">

    <include relativeToChangelogFile="true" file="../common-properties.xml"/>

    <changeSet id="changelog-event_store-1.2.0-timestamp" author="eurotech" dbms="mysql, mariadb" context="fixTimestamps">
        <modifyDataType tableName="sys_event_store" columnName="created_on" newDataType="timestamp(3) DEFAULT ${now}"/>
        <modifyDataType tableName="sys_event_store" columnName="modified_on" newDataType="timestamp(3) DEFAULT ${now}"/>
        <modifyDataType tableName="sys_event_store" columnName="event_on" newDataType="timestamp(3) DEFAULT ${now}"/>
    </changeSet>

</databaseChangeLog>

### liquibase excute has a exception.   timestamp(3)
When i modify timestamp(3) to timestamp and is ok.

Below is the exception info:

Reason: liquibase.exception.DatabaseException: Invalid default value for 'created_on' [Failed SQL: ALTER TABLE meeting.sys_event_store MODIFY created_on timestamp(3) DEFAULT current_timestamp()]
	at liquibase.changelog.ChangeSet.execute(ChangeSet.java:637)
	at liquibase.changelog.visitor.UpdateVisitor.visit(UpdateVisitor.java:53)
	at liquibase.changelog.ChangeLogIterator.run(ChangeLogIterator.java:83)
	at liquibase.Liquibase.update(Liquibase.java:202)
	at liquibase.Liquibase.update(Liquibase.java:179)
	at liquibase.Liquibase.update(Liquibase.java:175)
	at liquibase.Liquibase.update(Liquibase.java:168)
	at org.eclipse.kapua.commons.liquibase.KapuaLiquibaseClient.executeMasters(KapuaLiquibaseClient.java:224)
	at org.eclipse.kapua.commons.liquibase.KapuaLiquibaseClient.executeMasters(KapuaLiquibaseClient.java:196)
	at org.eclipse.kapua.commons.liquibase.KapuaLiquibaseClient.update(KapuaLiquibaseClient.java:149)
	at org.eclipse.kapua.app.api.web.RestApiListener.contextInitialized(RestApiListener.java:59)
	at org.apache.catalina.core.StandardContext.listenerStart(StandardContext.java:4579)
	at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5041)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
	at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:742)
	at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:718)
	at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:703)
	at org.apache.catalina.startup.HostConfig.deployWAR(HostConfig.java:987)
	at org.apache.catalina.startup.HostConfig$DeployWar.run(HostConfig.java:1858)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at org.apache.tomcat.util.threads.InlineExecutorService.execute(InlineExecutorService.java:75)
	at java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:112)
	at org.apache.catalina.startup.HostConfig.deployWARs(HostConfig.java:772)
	at org.apache.catalina.startup.HostConfig.deployApps(HostConfig.java:426)
	at org.apache.catalina.startup.HostConfig.start(HostConfig.java:1586)
	at org.apache.catalina.startup.HostConfig.lifecycleEvent(HostConfig.java:308)
	at org.apache.catalina.util.LifecycleBase.fireLifecycleEvent(LifecycleBase.java:123)
	at org.apache.catalina.util.LifecycleBase.setStateInternal(LifecycleBase.java:424)
	at org.apache.catalina.util.LifecycleBase.setState(LifecycleBase.java:367)
	at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:966)
	at org.apache.catalina.core.StandardHost.startInternal(StandardHost.java:839)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1427)
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1417)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at org.apache.tomcat.util.threads.InlineExecutorService.execute(InlineExecutorService.java:75)
	at java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:134)
	at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:943)
	at org.apache.catalina.core.StandardEngine.startInternal(StandardEngine.java:258)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
	at org.apache.catalina.core.StandardService.startInternal(StandardService.java:422)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
	at org.apache.catalina.core.StandardServer.startInternal(StandardServer.java:770)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
	at org.apache.catalina.startup.Catalina.start(Catalina.java:682)

**To Reproduce**
1. update to 1.2.0
2. update the database

**Expected behavior**

**Version of Kapua**
[1.2.0]

**Type of deployment**
[*] Local Vagrant deployment
[ ] Docker
[ ] Openshift (in its variants) 
[ ] Others

**Main component affected**
[ ] Console (in case of console please report info on which browser you encountered the problem)
[ ] REST API
[ ] Message Broker
[ *] - Others

### PR DESCRIPTION
**Pull request by miricy**
Modify the timestamp default value to current_timestamp(3) in mysql

In the file liquibase/common-properties.xml:
modify:
```
<property name="now" value="current_timestamp()" dbms="mysql,mariadb,h2"/>
```
to
```
<property name="now" value="current_timestamp(3)" dbms="mysql,mariadb,h2"/>
```
to fix #2991

### CODE REVIEW

File: commons/src/main/resources/liquibase/common-properties.xml [hunk 1]
```diff
@@ -17,5 +17,5 @@
 
     <property name="now" value="sysdate" dbms="oracle"/>
     <property name="now" value="now()" dbms="postgresql"/>
-    <property name="now" value="current_timestamp()" dbms="mysql,mariadb,h2"/>
+    <property name="now" value="current_timestamp(3)" dbms="mysql,mariadb,h2"/>
```
Comment (LeoNerdoG) [inline commons/src/main/resources/liquibase/common-properties.xml:21]: Pleae change the header to:

File: service/account/internal/src/main/resources/liquibase/common-properties.xml [hunk 2]
```diff
@@ -17,5 +17,5 @@
 
     <property name="now" value="sysdate" dbms="oracle"/>
     <property name="now" value="now()" dbms="postgresql"/>
-    <property name="now" value="current_timestamp()" dbms="mysql,mariadb,h2"/>
+    <property name="now" value="current_timestamp(3)" dbms="mysql,mariadb,h2"/>
```
Comment (LeoNerdoG) [inline service/account/internal/src/main/resources/liquibase/common-properties.xml:21]: Pleae change the header to:

File: service/device/registry/internal/src/main/resources/liquibase/common-properties.xml [hunk 3]
```diff
@@ -17,5 +17,5 @@
 
     <property name="now" value="sysdate" dbms="oracle"/>
     <property name="now" value="now()" dbms="postgresql"/>
-    <property name="now" value="current_timestamp()" dbms="mysql,mariadb,h2"/>
+    <property name="now" value="current_timestamp(3)" dbms="mysql,mariadb,h2"/>
```
Comment (LeoNerdoG) [inline service/device/registry/internal/src/main/resources/liquibase/common-properties.xml:21]: Pleae change the header to:

File: service/endpoint/internal/src/main/resources/liquibase/common-properties.xml [hunk 4]
```diff
@@ -17,5 +17,5 @@
 
     <property name="now" value="sysdate" dbms="oracle"/>
     <property name="now" value="now()" dbms="postgresql"/>
-    <property name="now" value="current_timestamp()" dbms="mysql,mariadb,h2"/>
+    <property name="now" value="current_timestamp(3)" dbms="mysql,mariadb,h2"/>
```
Comment (LeoNerdoG) [inline service/endpoint/internal/src/main/resources/liquibase/common-properties.xml:21]: Pleae change the header to:

File: service/tag/internal/src/main/resources/liquibase/common-properties.xml [hunk 5]
```diff
@@ -17,5 +17,5 @@
 
     <property name="now" value="sysdate" dbms="oracle"/>
     <property name="now" value="now()" dbms="postgresql"/>
-    <property name="now" value="current_timestamp()" dbms="mysql,mariadb,h2"/>
+    <property name="now" value="current_timestamp(3)" dbms="mysql,mariadb,h2"/>
```
Comment (LeoNerdoG) [inline service/tag/internal/src/main/resources/liquibase/common-properties.xml:21]: Pleae change the header to:

File: service/user/internal/src/main/resources/liquibase/common-properties.xml [hunk 6]
```diff
@@ -17,5 +17,5 @@
 
     <property name="now" value="sysdate" dbms="oracle"/>
     <property name="now" value="now()" dbms="postgresql"/>
-    <property name="now" value="current_timestamp()" dbms="mysql,mariadb,h2"/>
+    <property name="now" value="current_timestamp(3)" dbms="mysql,mariadb,h2"/>
```
Comment (LeoNerdoG) [inline service/user/internal/src/main/resources/liquibase/common-properties.xml:21]: Pleae change the header to:

**Review by LeoNerdoG — CHANGES_REQUESTED** [review]
Hey @miricy,
there are some headers to be changed. Please take a look.

**Comment by miricy** [comment]
@LeoNerdoG Do you mean to complete the title and description， Whats the header to be change

**Comment by LeoNerdoG** [comment]
Oh, I can see just now that Github did not parse my message correctly. 
Please change the year of the last modification of the file, so instead of only "2017", let it be "2017, 2020" (the first year is the year of the initial commit and the latter one is the year of the last change).

**Comment by Coduz** [comment]
Hi @miricy ,

thanks for the submission. 
I've tested this against also other DB and is working fine.
You may want to add your name to the Contributor in the header of the files? 

Like:
```
/*******************************************************************************
 * Copyright (c) 2017, 2020 Eurotech and/or its affiliates and others
 *
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     Eurotech - initial API and implementation
 *     <your name> - <your email>
 *******************************************************************************/
```

Regards, 

_Alberto_

**Comment by miricy** [comment]
Hi @Coduz  
Well, I've learned too much from this project, and I'm very happy to be able to help a little bit with this project。

**Comment by Coduz** [comment]
Hi @miricy ,

We also like contributions! 
Thanks! 

This will be merged now and will be backported in `release-1.2.x` and released with  Kapua `1.2.5`.

Regards, 

_Alberto_

**Pull request merged**